File: SoalA/memory.java
public class memory extends hardware {
    /* class memory adalah child dari class hardware */

    /* atribut privat */
    private int frequency;
    private int memorySize;
    private String supportsCuda;

    /* membuat constructor */
    public memory(int frequency, int memorySize, String supportsCuda) {
        this.frequency = frequency;
        this.memorySize = memorySize;
        this.supportsCuda = supportsCuda;
    }

    public memory() {

    }

    /* Get-Set Methods frequency */
    public void setFrequency(int frequency) {
        this.frequency = frequency;
    }

    public int getFrequency() {
        return frequency;
    }

    /* Get-Set Methods memorySize */
    public void setMemory(int memorySize) {
        this.memorySize = memorySize;
    }

    public int getMemory() {
        return memorySize;
    }

    /* Get-Set Methods supportsCuda */
    public void setCuda(String supportsCuda) {
        this.supportsCuda = supportsCuda;
    }

    public String getCuda() {
        return supportsCuda;
    }
}

File: SoalB/Pyhton/memory.py
from hardware import hardware


class memory(hardware):
    # class memory adalah child dari class hardware

    # atribut privat
    __frequency = 0
    __memorySize = 0
    __supportsCuda = ""

    # membuat constructor
    def __init__(self):
        self.__frequency = 0
        self.__memorySize = 0
        self.__supportsCuda = ""

    # get-set methods frequency
    def setFrequency(self, frequency):
        self.__frequency = frequency

    def getFrequency(self):
        return self.__frequency

    # get-set methods memorySize
    def setMemory(self, memorySize):
        self.__memorySize = memorySize

    def getMemory(self):
        return self.__memorySize

    # get-set methods supportsCuda
    def setCuda(self, supportsCuda):
        self.__supportsCuda = supportsCuda

    def getCuda(self):
        return self.__supportsCuda
